### PR TITLE
Development environment fixes

### PR DESCRIPTION
**Pull request by GTVolk**
Hello, i'am fix sonarlint errors about equality, added cross-env for Windows development, ability to comile sources with sourcemaps in production mode with environment variable SOURCEMAP, added different build and startup scripts.

P.S.: REQUIRES REVIEW: Added localStorage token caching for repeatable requests

### CODE REVIEW

File: src/scripts/http.js [hunk 1]
```diff
@@ -66,6 +66,7 @@ export class Http {
                 req.setRequestHeader(key, self._headers[key]);
               }
               if (bearer && bearer.token) {
+                localStorage.setItem("request-token", bearer.token);
```
Comment (Joxit) [inline src/scripts/http.js:69]: This may be a security issue https://stormpath.com/blog/where-to-store-your-jwts-cookies-vs-html5-web-storage

**Review by Joxit — CHANGES_REQUESTED** [review]
You want to add cross-env for Windows development right ? cross-env still works well, but is in maintenance mode, that's why I do not want to add this package to my project. I will propose a better solution and close your PR

File: package.json [hunk 2]
```diff
@@ -2,8 +2,15 @@
...
-    "build": "rollup -c",
+    "serve": "cross-env NODE_ENV=development ROLLUP_SERVE=true rollup -c -w",
+    "start:dev": "cross-env NODE_ENV=development rollup -c -w",
+    "start:prod": "cross-env NODE_ENV=production rollup -c -w",
+    "start": "npm run start:prod",
+    "start:sources": "cross-env SOURCEMAP=true npm run start:prod",
+    "build:dev": "cross-env NODE_ENV=development rollup -c",
+    "build:prod": "cross-env NODE_ENV=production rollup -c",
+    "build": "npm run build:prod",
+    "build:sources": "cross-env SOURCEMAP=true npm run build:prod",
```
[the remark this answers is not in the document]
Comment (GTVolk) [reply package.json:13]: Ok) I think these will be userful, i've use that for debugging and local development using your code :) I can remove it from this PR

File: package.json [hunk 3]
```diff
@@ -25,6 +32,7 @@
     "@rollup/plugin-json": "^4.1.0",
     "@rollup/plugin-node-resolve": "^13.1.3",
     "core-js": "^3.19.1",
+	"cross-env": "^7.0.3",
```
[the remark this answers is not in the document]
Comment (GTVolk) [reply package.json:35]: Ouch, CRLF to LF issues seems, i've fix that, thanks 👍

File: src/scripts/http.js (hunk 1 above)
Comment (GTVolk) [reply src/scripts/http.js:69]: Yeah, i know, but i need discussion here with you. Seems, when CORS is happen access token not stored in cookies in total, so, every request will be 401, and then request to auth in SSO, and then request to resource again. Imagine you have 50 image tags on front, fetching all of them without caching the key causes 150 requests :( You can check that in example

**Comment by GTVolk** [comment]
> You want to add cross-env for Windows development right ? cross-env still works well, but is in maintenance mode, that's why I do not want to add this package to my project. I will propose a better solution and close your PR

Yeah i want it too because unable to develop on Windows machine :( You know better alternative?

**Comment by Joxit** [comment]
Rrollup has an option `--environment`, see ab12cceefc5beb2829318ebefcb85e4ee4073043. Now the project should work on windows too

**Comment by GTVolk** [comment]
> Rrollup has an option `--environment`, see [ab12cce](https://github.com/Joxit/docker-registry-ui/commit/ab12cceefc5beb2829318ebefcb85e4ee4073043). Now the project should work on windows too

Wow, i don't know rollup so much, thank you!

**Pull request closed without merging**